src/Twilio/Clients/BearerToken/TwilioBearerTokenRestClient.cs

```diff
@@ -0,0 +1,317 @@
+﻿
+using System;
+using System.Net;
+using System.Linq;
+using Newtonsoft.Json;
+using Twilio.Exceptions;
+using Twilio.Http.BearerToken;
+using Twilio.Jwt;
+
+
+#if !NET35
+using System.Threading.Tasks;
+#endif
+
+using Twilio.Http;
+using Twilio.Http.BearerToken;
+#if NET35
+using Twilio.Http.Net35;
+#endif
+
+
+namespace Twilio.Clients.BearerToken
+{
+    /// <summary>
+    /// Implementation of a TwilioRestClient.
+    /// </summary>
+    public class TwilioBearerTokenRestClient
+    {
+        /// <summary>
+        /// Client to make HTTP requests
+        /// </summary>
+        public BearerTokenHttpClient HttpClient { get; }
+
+        /// <summary>
+        /// Twilio region to make requests to
+        /// </summary>
+        public string Region { get; }
```

Comment:
Customer has to set region and Edge for each client, try to check how can we improve, so that customer only need to set Region and Edge only once and it can be re-used for each client.